### DISCUSSION TITLE
agent_config.yaml

### DISCUSSION THREAD

**Discussion by ncd1992** [open-telemetry/opentelemetry-collector#5077] [Q&A] [answered]
Hello is it possible to use a base64 encoded password instead of clear test in the config file under receivers ?

**Comment by jpkrohling** [comment]
Why would you do that? It can't be for security, as base64 is as easy to decode as it is to encode.

**Comment by ncd1992** [comment]


**Reply by jpkrohling** [reply]
Base 64 is just an encoding, not encryption. I'll let others chime in on the Windows part.

**Comment by pmm-sumo** [comment, marked as answer]
I think that a better approach than storing password in the config would be passing it via an [environment variable](https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables) (I believe the expansion will work for Windows too)

**Reply by ncd1992** [reply]
Thankyou for the suggestion @pmm-sumo, I will try this out. If you have any examples for reference please do share that'll be helpful as well. I will keep this thread posted.

**Reply by jpkrohling** [reply]
It would likely be better than having it in the config file, but make sure it's an env var set for this process, not for the system. Otherwise, other processes can get access to the credentials.
